joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller_impl.h

```diff
@@ -393,48 +347,100 @@ update(const ros::Time& time, const ros::Duration& period)
   // fetch the currently followed trajectory, it has been updated by the non-rt thread with something that starts in the
   // next control cycle, leaving the current cycle without a valid trajectory.
 
-  // Update desired state: sample trajectory at current time
-  typename Trajectory::const_iterator segment_it = sample(curr_traj, time_data.uptime.toSec(), desired_state_);
-  if (curr_traj.end() == segment_it)
-  {
-    // Non-realtime safe, but should never happen under normal operation
-    ROS_ERROR_NAMED(name_,
-                    "Unexpected error: No trajectory defined at current time. Please contact the package maintainer.");
-    return;
-  }
+  std::vector<bool> successful_joint_traj;
+  successful_joint_traj.resize(joints_.size(), false);
```

Comment:
Dynamic memory allocations are disallowed in the real-time path.

In controllers, computation takes place in two places: 
- `update` method: contains periodic, real-time computation.
- ROS API callbacks: contains asynchronous, non real-time computation.

When implementing the `update` method, one has to be very careful not to break determinism and real-time friendliness.